src/librustc_mir/borrow_check/nll/renumber.rs

```diff
@@ -118,6 +118,11 @@ impl<'a, 'gcx, 'tcx> MutVisitor<'tcx> for NLLVisitor<'a, 'gcx, 'tcx> {
         debug!("visit_closure_substs: substs={:?}", substs);
     }
 
+    fn visit_user_assert_ty(&mut self, _ty: &mut Ty<'tcx>, _local: &mut Local,
+                            _location: Location) {
+        debug!("visit_user_assert_ty: skipping renumber");
```

Comment:
It'd be nice to have a comment here. Something like:

---

User-assert-ty statements represent types that the user added explicitly.
We don't want to erase the regions from these types: rather, we want to
add them as constraints at type-check time.

Reply:
Will add this as soon as I get a chance.